optuna_dashboard/static/components/GraphEdf.tsx

```diff
@@ -0,0 +1,80 @@
+import * as plotly from "plotly.js-dist"
+import React, { FC, useEffect } from "react"
+
+const plotDomId = "graph-edf"
+
+export const Edf: FC<{
+  trials: Trial[]
+}> = ({ trials = [] }) => {
+  useEffect(() => {
+    plotEdf(trials) // TODO(chenghuzi): Support multi-objective studies.
+  }, [trials])
+  return <div id={plotDomId} />
+}
+
+const plotEdf = (trials: Trial[]) => {
+  // Notice that this implementation is only for single study case
+  // as it's designed for single study details.
+  if (document.getElementById(plotDomId) === null) {
+    return
+  }
+  if (trials.length === 0) {
+    plotly.react(plotDomId, [])
+    return
+  }
+
+  const target_name = "Objective Value"
+
+  const _target = (t: Trial): number => {
+    return t.values![0]
+  }
+
+  const target = _target
```

Comment:
It looks that you can remove `_target` variable here.

Reply:
Yeah, but I added it here to remind me of a potential future multi-objective case, in which we may need to select between different `_target`s. So could we leave it as it is now?

Reply:
OK! I see.